src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -0,0 +1,161 @@
+    static propTypes = {
+        itemCollection: PropTypes.arrayOf(
+            PropTypes.shape({
+                type: PropTypes.string.isRequired,
+                absolute_path: PropTypes.string.isRequired,
+                name: PropTypes.string.isRequired,
+                modified_at: PropTypes.string.isRequired,
+                size: PropTypes.number.isRequired,
+                path_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+                parent: PropTypes.string,
+                item_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string.isRequired,
+                            name: PropTypes.string,
+                        }),
+                    ).isRequired,
+                }).isRequired,
+            }),
+        ).isRequired,
+    };
+
+    buildCache = memoize(itemList => {
+        const folders = [];
+        const temp = {};
+        const cache = {};
+        let root;
+
+        itemList.forEach(info => {
+            if (!info.parent) {
+                root = info.absolute_path;
+            }
+            if (info.type === 'folder') {
+                folders.push(info);
+            }
+            temp[info.absolute_path] = info;
+        });
+
+        folders.forEach(folderInfo => {
+            cache[folderInfo.absolute_path] = folderInfo.item_collection.entries.map(item => temp[item.absolute_path]);
+        });
+
+        return { cache, root };
+    });
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            fullPath: undefined,
+        };
+    }
+
+    getRowData = itemList => ({ index }) => {
+        const { name, type, modified_at: modifiedAt, ...rest } = itemList[index];
+
+        const rowData = {
+            [KEY_NAME]: {
+                isExternal: false,
+                name,
+                type,
+            },
+            [KEY_MODIFIED_AT]: `20${modifiedAt}`,
+            ...rest,
+        };
+        return rowData;
+    };
+
+    handleClick = cellValue => {
+        const { name } = cellValue;
+        const { fullPath } = this.state;
+        const nextFullPath = `${fullPath}${name}/`;
+        this.setState({
+            fullPath: nextFullPath,
+        });
+    };
+
+    render() {
+        const { itemCollection } = this.props;
+        const { fullPath } = this.state;
+        const { cache, root } = this.buildCache(itemCollection);
```

Comment:
Do we need to build the entire cache or can we find the correct `absolute_path` and return the related item list immediately? We should also separate the logic for finding the current item list and the root path into separate methods.